### PR TITLE
DOC: Fix examples in randint description

### LINKED ISSUE #18643: DOC: Problem with the randint description
### Issue with current documentation:

In the randint page https://scipy.github.io/devdocs/reference/generated/scipy.stats.randint.html

it is said that the index k can go from [low to high-1] (both included), so in that example from 7 to 30. The PMF values are calculated and shown correctly, but the generated k values are missing the value 30, i.e. only k values from 7 to 29 are generated and shown in the plot, when the value 30 should also be included.

### Idea or request for content:

One solution would be to change
`x = np.arange(randint.ppf(0.01, low, high),randint.ppf(0.99, low, high))`
for
`x = np.arange(low,high)`

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by Kyle0**
#### Reference issue
Closes #18643

#### What does this implement/fix?
Fixes x-axis range of example illustrating randint distribution

### CODE REVIEW

File: scipy/stats/_discrete_distns.py [hunk 1]
```diff
@@ -1051,7 +1051,42 @@ class randint_gen(rv_discrete):
...
+    >>> low, high = 7, 31
+    >>> mean, var, skew, kurt = randint.stats(low, high, moments='mvsk')
+
+    Display the probability mass function (pmf):
+
+    >>> x = np.arange(low - 5, high + 5)
+    >>> ax.plot(x, randint.pmf(x, low, high), 'bo', ms=8, label='randint pmf')
+    >>> ax.vlines(x, 0, randint.pmf(x, low, high), colors='b', lw=5, alpha=0.5)
+    
+    Alternatively, the distribution object can be called (as a function) to fix the shape and location. This returns a “frozen” RV object holding the given parameters fixed.
```
Comment (mdhaber) [inline scipy/stats/_discrete_distns.py:1072]:
```suggestion
    Alternatively, the distribution object can be called (as a function) to
    fix the shape and location. This returns a "frozen" RV object holding the
    given parameters fixed.
```

File: scipy/stats/_discrete_distns.py [hunk 2]
```diff
@@ -1051,7 +1051,42 @@ class randint_gen(rv_discrete):
...
+
+    Freeze the distribution and display the frozen pmf:
+
+    >>> rv = randint(low, high)
+    >>> ax.vlines(x, 0, rv.pmf(x), colors='k', linestyles='-', lw=1,
+    ...         label='frozen pmf')
+    >>> ax.legend(loc='best', frameon=False)
```
Comment (mdhaber) [inline scipy/stats/_discrete_distns.py:1079]: Please move the legend off to the side while you're at it:
image

**Comment by mdhaber** [comment]
Thanks @Kyle0; I'll push a commit to fix the CI failure and the issue I mentioned, then I'll merge.
Note that in the future, you can include `[skip actions] [skip cirrus]` to ensure that only the documentation CI jobs are run, since this has no changes to the code.

**Comment by Kyle0** [comment]
Thanks mdhaber

File: scipy/stats/_discrete_distns.py [hunk 3]
```diff
@@ -1051,7 +1051,47 @@ class randint_gen(rv_discrete):
...
+    Alternatively, the distribution object can be called (as a function) to 
+    fix the shape and location. This returns a "frozen" RV object holding the
+    given parameters fixed.
+
+    Freeze the distribution and display the frozen ``pmf``:
+
+    >>> rv = randint(low, high)
+    >>> ax.vlines(x, 0, rv.pmf(x), colors='k', linestyles='-',
+    ...           lw=1, label='frozen pmf')
+    >>> ax.legend(loc='upper right', bbox_to_anchor=(1.3, 1))
```
Comment (mdhaber) [inline scipy/stats/_discrete_distns.py:1081]:
```suggestion
    >>> ax.legend(loc='lower center')
```

**Pull request merged**